src/openassetio-core-python/python/hostApi.cpp

```diff
@@ -0,0 +1,47 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2022 The Foundry Visionmongers Ltd
+#include <openassetio/python/hostApi.hpp>
+
+#include <memory>
+
+#include <pybind11/embed.h>
+
+#include <openassetio/LoggerInterface.hpp>
+#include <openassetio/hostApi/ManagerImplementationFactoryInterface.hpp>
+
+namespace py = pybind11;
+
+namespace openassetio {
+inline namespace OPENASSETIO_CORE_ABI_VERSION {
+namespace python::hostApi {
+
+using openassetio::hostApi::ManagerImplementationFactoryInterface;
+using openassetio::hostApi::ManagerImplementationFactoryInterfacePtr;
+
+ManagerImplementationFactoryInterfacePtr createPluginSystemManagerImplementationFactory(
+    LoggerInterfacePtr logger) {  // NOLINT(performance-unnecessary-value-param)
+  // Get Python class.
+  py::object pyClass =
+      py::module_::import("openassetio.pluginSystem.PluginSystemManagerImplementationFactory")
+          .attr("PluginSystemManagerImplementationFactory");
+
+  // Instantiate Python object.
+  py::object pyInstance = pyClass(std::move(logger));
+
+  // Keep Python object alive whilst shared_ptr chain is alive.
+  auto pyInstancePtr =
+      std::shared_ptr<py::object>{new py::object{pyInstance}, [](py::object* pyObjectPtr) {
+                                    py::gil_scoped_acquire gil;
+                                    delete pyObjectPtr;
+                                  }};
```

Comment:
[consider] Is there any way we can move this to a helper in `PyRetaningSharedPtr.hpp` as it's the same as the inner code in the `type_caster`, and it's well documented what it's upto there.

Reply:
Split out in 09f09b2a (note: juggled commits and force-pushed).